### PR TITLE
MARTUSDEV-1890: Improve exception handling in load data tasks

### PR DESCRIPTION
**Pull request by pmuchowski**

### CODE REVIEW

File: martus-client/source/org/martus/client/swingui/jfx/landing/general/ManageServerSyncRecordsController.java [hunk 1]
```diff
@@ -209,6 +214,16 @@ protected void succeeded()
 
 			serverMyDrafts = getValue();
 		}
+
+		@Override
+		protected void failed()
```
Comment (nimbooce) [inline martus-client/source/org/martus/client/swingui/jfx/landing/general/ManageServerSyncRecordsController.java]: The failed methods can be extracted into the main class and and reused by all three tasks.

Comment (nimbooce) [reply martus-client/source/org/martus/client/swingui/jfx/landing/general/ManageServerSyncRecordsController.java]: Pass in the message

Comment (pmuchowski) [reply martus-client/source/org/martus/client/swingui/jfx/landing/general/ManageServerSyncRecordsController.java]: @nimbooce PR updated

File: martus-client/source/org/martus/client/swingui/jfx/landing/general/ManageServerSyncRecordsController.java [hunk 2]
```diff
@@ -159,6 +160,9 @@ protected SyncRecordsTableProvider call() throws Exception
 				hqDraftsThread.join();
 				hqSealedsThread.join();
 
+				if (loadRecordsError)
+					throw new RuntimeException("Failed to load data");
```
Comment (nimbooce) [inline martus-client/source/org/martus/client/swingui/jfx/landing/general/ManageServerSyncRecordsController.java]: Should throw Exception

File: martus-client/source/org/martus/client/swingui/jfx/landing/general/ManageServerSyncRecordsController.java [hunk 3]
```diff
@@ -193,7 +198,18 @@ private Thread createAndStartThread(Task task)
...
 
-	protected class ServerMyDraftsTask extends Task<Vector>
+	protected abstract class LoadServerData extends Task<Vector>
+	{
+		protected void dataLoadFailed(String message)
+		{
+			loadRecordsError = true;
+
+			if (getException() != null)
+				MartusLogger.logException(new RuntimeException(message, getException()));
```
Comment (nimbooce) [inline martus-client/source/org/martus/client/swingui/jfx/landing/general/ManageServerSyncRecordsController.java]: Create a new Exception.

**Pull request merged**